dll/win32/browseui/shellbrowser.cpp

```diff
@@ -2459,12 +2460,46 @@ HRESULT STDMETHODCALLTYPE CShellBrowser::GetTravelLog(ITravelLog **pptl)
 
 HRESULT STDMETHODCALLTYPE CShellBrowser::ShowControlWindow(UINT id, BOOL fShow)
 {
+    BOOL shown;
+    if (FAILED(IsControlWindowShown(id, &shown)))
+        return E_NOTIMPL;
+    else if (!shown == !fShow) // Negated for true boolean comparison
+        return S_OK;
+    else switch (id)
+    {
+        case FCW_STATUS:
+            OnToggleStatusBarVisible(0, 0, NULL, shown);
+            return S_OK;
+        case FCW_TREE:
+            return Exec(&CGID_Explorer, SBCMDID_EXPLORERBARFOLDERS, 0, NULL, NULL);
+    }
     return E_NOTIMPL;
 }
 
 HRESULT STDMETHODCALLTYPE CShellBrowser::IsControlWindowShown(UINT id, BOOL *pfShown)
 {
-    return E_NOTIMPL;
+    OLECMD cmd = {};
+    HRESULT hr = S_OK;
+    BOOL shown = FALSE;
+    switch (id)
+    {
+        case FCW_STATUS:
+            shown = m_settings.fStatusBarVisible;
+            break;
+        case FCW_TREE:
+            cmd.cmdID = SBCMDID_EXPLORERBARFOLDERS;
+            hr = QueryStatus(&CGID_Explorer, 1, &cmd, NULL);
+            shown = cmd.cmdf & OLECMDF_LATCHED;
+            break;
+        default:
+            hr = E_NOTIMPL;
+    }
+    if (pfShown)
+    {
+        *pfShown = shown;
+        return hr;
```

Comment:
Not using the code below instead? why?

Reply:
Because it is unrelated!

If the caller passes in a non-NULL pfShown then the show state ends up there and the return is S_OK or E_*.
With a null pfShown S_OK vs S_FALSE is the show state.

Reply:
thanks for the explanation :+1: